src/openms_gui/source/VISUAL/SpectraIdentificationViewWidget.cpp

```diff
@@ -488,202 +386,60 @@ namespace OpenMS
       // skip
       if (hide_no_identification_->isChecked() && id_count == 0)  { continue; }
 
-      // set row color
-      QColor c;
-      if (id_count == 0)
-      {
-        c = Qt::white; // no identification
-      }
-      else
-      {
-        c = Qt::green; // identification
-      }
+      // set row background color
+      QColor bg_color = (id_count == 0 ? bg_color = Qt::white : bg_color = Qt::green);
```

Comment:
Are the assignments in the cond. stmts needed?

Reply:
no. good catch. this only works because the assignment returns a reference. But its weird, so I'll remove it.